### PR TITLE
feat(swift): add support for cocoapods lock files

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add support for swift [cocoapods](https://cocoapods.org/) lock files.

### CODE REVIEW

File: pkg/swift/cocoapods/parse.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	decoder := yaml.NewDecoder(r)
+	if err := decoder.Decode(&lock); err != nil {
+		return nil, nil, xerrors.Errorf("failed to decode cocoapods lock file: %s", err.Error())
+	}
+
+	parsedDeps := map[string]types.Library{} // dependency name => Library
+	directDeps := map[string][]string{}      // dependency name => slice of child dependency names
+	for _, pod := range lock.Pods {
+		switch reflect.ValueOf(pod).Kind() {
+		case reflect.String: // dependency with version number
```
Comment (knqyf263) [inline pkg/swift/cocoapods/parse.go:40]:
```suggestion
		switch pod.(type) {
		case string:
```

File: pkg/swift/cocoapods/parse.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		}
+	}
+
+	var deps []types.Dependency
+	for dep, childDeps := range directDeps {
+		var dependsOn []string
+		// find versions for direct dependencies
+		for _, childDep := range childDeps {
+			// mark this dep as indirect
+			lib := parsedDeps[childDep]
```
Comment (knqyf263) [inline pkg/swift/cocoapods/parse.go:74]:
```
  - AppCenter (4.2.0):
    - AppCenter/Analytics (= 4.2.0)
    - AppCenter/Crashes (= 4.2.0)
  - AppCenter/Analytics (4.2.0):
    - AppCenter/Core
```

In this example, `AppCenter/Analytics` can be direct dependency, no? IIUC, it just says `AppCenter` depends on `AppCenter/Analytics`, but we're not sure if `AppCenter/Analytics` is used directly or not. Please correct me if I'm wrong.

Comment (DmitriyLewen) [reply pkg/swift/cocoapods/parse.go:74]: hm... You are right. I didn't thinks about this.
I will change it.

File: pkg/swift/cocoapods/parse.go (hunk 1 above)
Comment (DmitriyLewen) [reply pkg/swift/cocoapods/parse.go]: Done

File: pkg/swift/cocoapods/parse.go (hunk 2 above)
Comment (DmitriyLewen) [reply pkg/swift/cocoapods/parse.go]: Changed

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**